.github/workflows/deploy-to-eks.yaml

```diff
@@ -2,21 +2,29 @@ name: Create EKS cluster, deploy CKF and run bundle test
 on:
   workflow_dispatch: # This event allows manual triggering from the Github UI
     inputs:
+      bundle_version:
+        description: 'Comma-separated list of bundle versions e.g. "1.7","1.8"'
+        default: '"1.7","1.8","latest"'
+        required: true
+      k8s_version:
+        description: 'Kubernetes version to be used for the AKS cluster'
+        required: false
       uats_branch:
         description: 'Branch to run the UATs from e.g. main or track/1.8. By default, the workflow uses main.'
         required: false
-    secrets:
-      BUNDLE_KUBEFLOW_EKS_AWS_ACCESS_KEY_ID:
-        required: true
-      BUNDLE_KUBEFLOW_EKS_AWS_SECRET_ACCESS_KEY:
-        required: true
   schedule:
     - cron: "23 0 * * 2"
 jobs:
   deploy-ckf-to-eks:
     runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        bundle_version: ${{ fromJSON(format('[{0}]', inputs.bundle_version || '"1.7","1.8","latest"')) }}
+      fail-fast: false
     env:
-      UATS_BRANCH: ${{ inputs.uats_branch || 'main' }}
+      K8S_VERSION: ${{ inputs.k8s_version || fromJSON('{"1.7":"1.24", "1.8":"1.26", "latest":"1.29"}')[matrix.bundle_version] }}
+      JUJU_VERSION: ${{ fromJSON('{"1.7":"2.9.49","1.8":"3.5.0","latest":"3.5.0"}')[ matrix.bundle_version ] }}
+      UATS_BRANCH: ${{ inputs.uats_branch || fromJSON('{"1.7":"track/1.7", "1.8":"main", "latest":"main"}')[matrix.bundle_version] }}
```

Comment:
Is there a way in which we can abstract these "maps" into an input or a default? I'm thinking about maintenance of this workflow, it'd be easier if we just had this information in one place that could be easier of us to update instead of having to look all the way through this file.

Reply:
Thanks for challenging me on this one. After looking into this, I created a `dependencies.yaml` file with the deps and used this action https://github.com/marketplace/actions/yaml-to-github-output for converting those to env variables.

Reply:
The proposal looks amazing! Thanks!